components/blitz/src/omero/gateway/facility/DataManagerFacility.java

```diff
@@ -226,35 +226,11 @@ public CmdCallbackI deleteFolders(SecurityContext ctx,
     public CmdCallbackI delete(SecurityContext ctx, List<IObject> objects)
             throws DSOutOfServiceException, DSAccessException {
         try {
-            /*
-             * convert the list of objects to lists of IDs by OMERO model class
-             * name
-             */
-            final Map<String, List<Long>> objectIds = new HashMap<String, List<Long>>();
+            final Delete2Builder request = Requests.delete();
             for (final IObject object : objects) {
-                /* determine actual model class name for this object */
-                Class<? extends IObject> objectClass = object.getClass();
-                while (true) {
-                    final Class<?> superclass = objectClass.getSuperclass();
-                    if (IObject.class == superclass) {
-                        break;
-                    } else {
-                        objectClass = superclass.asSubclass(IObject.class);
-                    }
-                }
-                final String objectClassName = objectClass.getSimpleName();
-                /* then add the object's ID to the list for that class name */
-                final Long objectId = object.getId().getValue();
-                List<Long> idsThisClass = objectIds.get(objectClassName);
-                if (idsThisClass == null) {
-                    idsThisClass = new ArrayList<Long>();
-                    objectIds.put(objectClassName, idsThisClass);
-                }
-                idsThisClass.add(objectId);
+                request.target(object);
```

Comment:
Alas no, thanks to Java dreadfully having poor-man's generics: the type erasure doesn't let the builder distinguish all the kinds of collection safely, and `Iterable<IObject>` was one of the alternatives I thus omitted. (Of course there are erasure issues with object arrays too.)